File: Backend/plated/src/main/java/com/priyanshparekh/plated/user/profile/UserProfileDto.java
package com.priyanshparekh.plated.user.profile;

import com.priyanshparekh.plated.recipe.dto.ProfileRecipeItem;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserProfileDto {

    private Long id;
    private String displayName;
    private String bio;
    private String profession;
    private String website;
    private int recipeCount;
    private int followersCount;
    private int followingCount;
    private List<ProfileRecipeItem> recipes;

}

File: Frontend/Plated2/core-model/src/main/java/com/priyanshparekh/core/model/profile/UserProfileDto.kt
package com.priyanshparekh.core.model.profile

import com.priyanshparekh.core.model.recipe.RecipeItem.ProfileRecipeItem

data class UserProfileDto(
    val displayName: String,
    val bio: String,
    val profession: String,
    val website: String,
    val recipeCount: Int,
    val followersCount: Int,
    val followingCount: Int,
    val recipes: List<ProfileRecipeItem>
)
